tests/contrib/redis/test.py

```diff
@@ -193,81 +119,48 @@ def test_patch_unpatch(self):
         assert spans, spans
         assert len(spans) == 1
 
+    @snapshot()
     def test_opentracing(self):
         """Ensure OpenTracing works with redis."""
-        ot_tracer = init_tracer("redis_svc", self.tracer)
+        ot_tracer = init_tracer("redis_svc", ddtrace.tracer)
 
         with ot_tracer.start_active_span("redis_get"):
             us = self.r.get("cheese")
             assert us is None
 
-        spans = self.get_spans()
-        assert len(spans) == 2
-        ot_span, dd_span = spans
-
-        # confirm the parenting
-        assert ot_span.parent_id is None
-        assert dd_span.parent_id == ot_span.span_id
-
-        assert ot_span.name == "redis_get"
-        assert ot_span.service == "redis_svc"
-
-        assert_is_measured(dd_span)
-        assert dd_span.service == "redis"
-        assert dd_span.name == "redis.command"
-        assert dd_span.span_type == "redis"
-        assert dd_span.error == 0
-        assert dd_span.get_metric("out.redis_db") == 0
-        assert dd_span.get_tag("out.host") == "localhost"
-        assert dd_span.get_tag("redis.raw_command") == u"GET cheese"
-        assert dd_span.get_metric("redis.args_length") == 2
-        assert dd_span.resource == "GET cheese"
-
     @BaseTracerTestCase.run_in_subprocess(env_overrides=dict(DD_SERVICE="mysvc"))
+    @snapshot()
```

Comment:
The snapshot here has to come before the `run_in_subprocess` which we should probably try to add a check for if possible in `@snapshot()`.

Reply:
`@run_in_subprocess(snapshot=True)` ?

not sure if it'll be too obscure of an option or not

Reply:
Hmm yeah, not sure about that one but it might save some headaches down the line for people copy-pasting test cases which would make it worth it IMO.